common/lib/common-utils/src/promises.ts

```diff
@@ -61,19 +59,6 @@ export class Deferred<T> {
     }
 }
 
-/**
- * Helper function that asserts that the given promise only resolves
- */
-// eslint-disable-next-line @typescript-eslint/promise-function-async
-export function assertNotRejected<T>(promise: Promise<T>): Promise<T> {
```

Comment:
Couldn't find any usages of this in our code. I was expecting tree shaking to get rid of this, but since assert is not tree shakeable/marked as side effect free, it still gets pulled in even if it is unused.